### PR TITLE
feat: use the new function to create a container within a pod

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR uses the updated createContainer that allows to create containers inside a pod. 
It also added the Z flag to avoid errors like in applehv 

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

it resolves https://github.com/projectatomic/ai-studio/issues/131

### How to test this PR?

1. just run a recipe
-  it should work as before on win and qemu
- it should work fine on applehv

### CODE REVIEW

File: packages/backend/src/utils/pathUtils.ts [hunk 1]
```diff
@@ -25,3 +26,11 @@ export function getParentDirectory(filePath: string): string {
   // Get the directory name using path.dirname
   return path.dirname(normalizedPath);
 }
+
+export function getMappedPathInPodmanMachine(path: string): string {
```
[the remark this answers is not in the document]
Comment (jeffmaury) [reply packages/backend/src/utils/pathUtils.ts:30]: I am a bit puzzled by this. I don't understand why we need to convert the path (as using the CLI I can specifiy a Windows path) where it was not required before. Isn't an issue on the libpod side ?

[the remark this answers is not in the document]
Comment (lstocchi) [reply packages/backend/src/utils/pathUtils.ts:30]: I guess so, if i start the app with the original path i face 
```
Error starting the application "ChatBot": Error: (HTTP code 500) server error - statfs /C:\Users\luca\podman-desktop\ai-studio\models\hf.TheBloke.llama-2-7b-chat.Q5_K_S\llama-2-7b-chat.Q5_K_S.gguf: no such file or directory
```

[the remark this answers is not in the document]
Comment (benoitf) [reply packages/backend/src/utils/pathUtils.ts:30]: it looks strange to give an internal name in the machine

If I look at 'podman create' command

If I do for example `podman create -v $(pwd):/foo fedora`

it's sending

```json
{
   "env_host":false,
   "httpproxy":true,
   "terminal":false,
   "stdin":false,
   "stop_timeout":10,
   "log_configuration":{
      
   },
   "systemd":"true",
   "sdnotifyMode":"container",
   "pidns":{
      
   },
   "utsns":{
      
   },
   "remove":false,
   "containerCreateCommand":[
      "podman",
      "create",
      "-v",
      "/Users/benoitf/my-test/foo",
      "fedora"
   ],
   "init_container_type":"",
   "unsetenvall":false,
   "manage_password":true,
   "image":"fedora",
   "raw_image_name":"fedora",
   "image_volume_mode":"anonymous",
   "init":false,
   "mounts":[
      {
         "destination":"/foo",
         "type":"bind",
         "source":"/Users/benoitf/my-test"
      }
   ],
   "ipcns":{
      
   },
   "volatile":false,
   "privileged":false,
   "seccomp_policy":"default",
   "userns":{
      
   },
   "idmappings":{
      "HostUIDMapping":true,
      "HostGIDMapping":true,
      "UIDMap":null,
      "GIDMap":null,
      "AutoUserNs":false,
      "AutoUserNsOpts":{
         "Size":0,
         "InitialSize":0,
         "PasswdFile":"",
         "GroupFile":"",
         "AdditionalUIDMappings":null,
         "AdditionalGIDMappings":null
      }
   },
   "read_only_filesystem":false,
   "read_write_tmpfs":false,
   "umask":"0022",
   "cgroupns":{
      
   },
   "netns":{
      
   },
   "publish_image_ports":false,
   "Networks":null,
   "use_image_resolve_conf":false,
   "use_image_hosts":false
}
```

interesting part is

```
"mounts":[
      {
         "destination":"/foo",
         "type":"bind",
         "source":"/Users/benoitf/my-test"
      }
```

which is the path on my machine/host, not inside the podman machine

[the remark this answers is not in the document]
Comment (benoitf) [reply packages/backend/src/utils/pathUtils.ts:30]: should it be reported to podman ? looks like client shouldn't need to do that, API should convert to /mnt/c

**Comment by lstocchi** [comment]
@jeffmaury rebased and removed the mapping function as not needed anymore

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**